modules/local/bwamem2_mem.nf

```diff
@@ -1,29 +1,38 @@
-params.bwa_options = "-K 100000000 -M"
-params.sequencer = "ILLUMINA"
-
 process BWAMEM2_MEM {
-    label 'CPUS_MAX'
-
-    tag "${sample}_${run}"
+    tag "${meta.id}"
+    label 'process_high'
 
-    publishDir "${params.outdir}/bwamem2_mem", mode: 'copy'
+    publishDir "${params.outdir}/bwamem2/${meta.sample}",
```

Comment:
You should remove any customisation from this code in terms of output directories as this should be customisable from the `opts` map that comes into the module. The logic still needs a little work but for now this is the generic code I am using [here](https://github.com/drpatelh/chipseq/blob/7bd0e6b0e619754845e2f65bb4641671bba42bdf/modules/nf-core/software/bwa_mem.nf#L4-L9)